docs/clients/promtail/configuration.md

```diff
@@ -0,0 +1,943 @@
+targets, see [Scraping](scraping.md). For more information on transforming logs
+from scraped targets, see [Pipelines](pipelines.md).
+
+Generic placeholders are defined as follows:
+
+* `<boolean>`: a boolean that can take the values `true` or `false`
+* `<int>`: any integer matching the regular expression `[1-9]+[0-9]*`
+* `<duration>`: a duration matching the regular expression `[0-9]+(ms|[smhdwy])`
+* `<labelname>`: a string matching the regular expression `[a-zA-Z_][a-zA-Z0-9_]*`
+* `<labelvalue>`: a string of unicode characters
+* `<filename>`: a valid path relative to current working directory or an
+    absolute path.
+* `<host>`: a valid string consisting of a hostname or IP followed by an optional port number
+* `<string>`: a regular string
+* `<secret>`: a regular string that is a secret, such as a password
+
+Supported contents and default values of `config.yaml`:
+
+```yaml
+# Configures the server for Promtail.
+[server: <server_config>]
+
+# Describes how promtail will connect to Loki
+# DEPRECATED. Use clients instead.
+[client: <client_config>]
+
+# Describes how promtail connects to multiple instances
+# of Loki, sending logs to each.
+clients:
+  - [<client_config>]
+
+# Describes how to save read file offsets to disk
+[positions: <position_config>]
+
+scrape_configs:
+  - [<scrape_config>]
+
+# Configures how tailed targets will be watched.
+[target_config: <target_config>]
+```
+
+## server_config
+
+The `server_config` block configures Promtail's behavior as an HTTP server:
+
+```yaml
+# HTTP server listen host
+[http_listen_host: <string>]
+
+# HTTP server listen port
+[http_listen_port: <int> | default = 80]
+
+# gRPC server listen host
+[grpc_listen_host: <string>]
+
+# gRPC server listen port
+[grpc_listen_port: <int> | default = 9095]
+
+# Register instrumentation handlers (/metrics, etc.)
+[register_instrumentation: <boolean> | default = true]
+
+# Timeout for graceful shutdowns
+[graceful_shutdown_timeout: <duration> | default = 30s]
+
+# Read timeout for HTTP server
+[http_server_read_timeout: <duration> | default = 30s]
+
+# Write timeout for HTTP server
+[http_server_write_timeout: <duration> | default = 30s]
+
+# Idle timeout for HTTP server
+[http_server_idle_timeout: <duration> | default = 120s]
+
+# Max gRPC message size that can be received
+[grpc_server_max_recv_msg_size: <int> | default = 4194304]
+
+# Max gRPC message size that can be sent
+[grpc_server_max_send_msg_size: <int> | default = 4194304]
+
+# Limit on the number of concurrent streams for gRPC calls (0 = unlimited)
+[grpc_server_max_concurrent_streams: <int> | default = 100]
+
+# Log only messages with the given severity or above. Supported values [debug,
+# info, warn, error]
+[log_level: <string> | default = "info"]
+
+# Base path to server all API routes from (e.g., /v1/).
+[http_path_prefix: <string>]
+```
+
+## client_config
+
+The `client_config` block configures how Promtail connects to an instance of
+Loki:
+
+```yaml
+# The URL of Loki.
```

Comment:
What exactly should this URL point to? URL of distributor HTTP API endpoint? Or GRPC?

Reply:
Good question. In single-process mode, it should be pointing at the endpoint defined by Loki's http_listen_host and http_listen_port and not the gRPC listen addresses. In multi-process mode, it should point to the distributor's http_listen_host and http_listen_port. I've updated the docs.

Reply:
I've figured it out after some searching, but it's good to have it documented for others. Thanks!